tds-itemselection-aironline/src/main/java/tds/itemselection/algorithms/AdaptiveSelector.java

```diff
@@ -108,6 +112,11 @@ public ItemGroup getNextItemGroup (SQLConnection connection,
     return result;
   }
 
+  @Override
+  public ItemGroup getNextItemGroup(SQLConnection connection, ItemCandidatesData itemData, List<ItemGroup> itemGroups) throws ItemSelectionException {
```

Comment:
Instead of adding this method and throwing `NotImplementedException` everywhere could we change the base method to take in a null/empty list and handle it accordingly?  If I wanted to use an API it would be odd for me to see a method, call it, and then see it throw.